tests/unit/_experimental/object_detection/test_dataset.py

```diff
@@ -210,9 +210,30 @@ def test__upload_object_detection_ignore_field() -> None:
             threshold_strategy="F1-Optimal",
             min_confidence_score=0.222,
         )
-    assert [ScoredBoundingBox(**elem) for elem in df["TP"][0]] == [
```

Comment:
this is the only test need to be fixed due to `dataclasses.field(init=False)` at https://github.com/kolenaIO/kolena/blob/0bbc4e5217b9ad76e36f5d73c12d3015057026e9/kolena/annotation.py#L86-L89, which fails like

```sh
ScoredBoundingBox(top_left=[2.0, 2.0], bottom_right=[3.0, 3.0], width=1.0, height=1.0, area=1.0, aspect_ratio=1.0, score=1.0)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-17-feec49ea9a42> in <cell line: 1>()
----> 1 ScoredBoundingBox(top_left=[2.0, 2.0], bottom_right=[3.0, 3.0], width=1.0, height=1.0, area=1.0, aspect_ratio=1.0, score=1.0)

~/Documents/kolena-repos/kolena/.venv/lib/python3.9/site-packages/pydantic/v1/dataclasses.py in handle_extra_init(self, *args, **kwargs)
    296             for k, v in kwargs.items():
    297                 self.__dict__.setdefault(k, v)
--> 298             init(self, *args, **{k: v for k, v in kwargs.items() if k in self.__dataclass_fields__})
    299 
    300         else:

TypeError: __init__() got an unexpected keyword argument 'width'
```

Reply:
should we create two environments to run tests, where one install pydantic v1 and one install pydantic v2?

Reply:
Does this mean we decided to stay with pydantic v1?

Reply:
shouldn't `extra="allow"` enable `ScoredBoudningBox` to take extra fields?

Reply:
> should we create two environments to run tests, where one install pydantic v1 and one install pydantic v2?

i think running unit test with pydantic v1 has different meaning with using kolena with pydantic v1 in an example. also setting this kind of testing matrix is cumbersome and slow
so i am inclined to manually verify it works.

 > Does this mean we decided to stay with pydantic v1?

de facto yes

> shouldn't extra="allow" enable ScoredBoudningBox to take extra fields? Asking to ensure the grabbag datatypes feature is not breaking.

`width` is kind of a reserved the field for this class based on `dataclasses.field(init=False)`, the grabbag still work for other non reserved fields

```sh
In [11]: from kolena.annotation import ScoredBoundingBox

In [12]: ScoredBoundingBox(top_left=[2.0, 2.0], bottom_right=[3.0, 3.0], score=1.0, random_field=True)
Out[12]: ScoredBoundingBox(top_left=(2.0, 2.0), bottom_right=(3.0, 3.0), width=1.0, height=1.0, area=1.0, aspect_ratio=1.0, score=1.0)

In [13]: bbox = ScoredBoundingBox(top_left=[2.0, 2.0], bottom_right=[3.0, 3.0], score=1.0, random_field=True)

In [14]: bbox.random_field
Out[14]: True
```

Reply:
Can we update the test with different field, but keeping using the dataclass, instead of verify through `dict`? Want to keep the test covering dataclass deserialization.

Reply:
i think the `TP` is a list of dictionary from `df = object_detection.dataset.compute_object_detection_results`, so i don't think there are any intentional dataclass deserialization testing conducted at here, it's just a convenient way to compare results, that's why we create the object using dictionary originally like `[ScoredBoundingBox(**elem) for elem in df["TP"][0]]`